router/network.go

```diff
@@ -28,19 +30,106 @@ import (
 
 var contentTypeRegex = regexp.MustCompile(`^(text/[a-z0-9.-]+)|(application/([a-z0-9.-]+\+)?(json|xml))$`)
 
+// Default private IP ranges in CIDR notation
+const defaultPrivateIPRanges = "10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,127.0.0.0/8,169.254.0.0/16,fc00::/7,fe80::/10"
+
+// IPRange represents a range of IP addresses
+type IPRange struct {
+	start net.IP
+	end   net.IP
+}
+
+// parseCIDRToRange converts a CIDR block to an IP range
+func parseCIDRToRange(cidr string) (IPRange, error) {
+	_, ipnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return IPRange{}, fmt.Errorf("invalid CIDR block %s: %w", cidr, err)
+	}
+
+	// Get the network address (start of range)
+	start := ipnet.IP
+
+	// Calculate the broadcast address (end of range)
+	mask := ipnet.Mask
+	end := make(net.IP, len(start))
+	for i := 0; i < len(start); i++ {
+		end[i] = start[i] | ^mask[i]
+	}
+
+	return IPRange{start: start, end: end}, nil
+}
+
+// loadPrivateIPRanges loads private IP ranges from environment variable or uses defaults
+func loadPrivateIPRanges(config *config.Config) []IPRange {
+	cidrBlocks := config.GetString("Router.privateIPRanges", defaultPrivateIPRanges)
+	if cidrBlocks == "" {
+		cidrBlocks = defaultPrivateIPRanges
+	}
+
+	var ranges []IPRange
+	for _, cidr := range strings.Split(cidrBlocks, ",") {
+		cidr = strings.TrimSpace(cidr)
+		if cidr == "" {
+			continue
+		}
+
+		ipRange, err := parseCIDRToRange(cidr)
+		if err != nil {
+			// Log error but continue with other ranges
+			fmt.Printf("Error parsing CIDR block %s: %v\n", cidr, err)
+			continue
+		}
+		ranges = append(ranges, ipRange)
+	}
+
+	return ranges
+}
+
```

Comment:
I believe that we just need a simlpe `rudder-go-kit` function, named `net.IsPrivateIP`
```go
var privateIPBlocks []*net.IPNet

func init() {
	for _, cidr := range []string{
		"10.0.0.0/8",
		"172.16.0.0/12",
		"192.168.0.0/16",
		"169.254.0.0/16",  // Link-local
		"127.0.0.0/8",     // Loopback
		"::1/128",         // IPv6 loopback
		"fc00::/7",        // IPv6 unique local address
		"fe80::/10",       // IPv6 link-local
	} {
		_, block, _ := net.ParseCIDR(cidr)
		privateIPBlocks = append(privateIPBlocks, block)
	}
}

func IsPrivateIP(ip net.IP) bool {
	for _, block := range privateIPBlocks {
		if block.Contains(ip) {
			return true
		}
	}
	return false
}
```

Reply:
This makes more sense.. I was wondering where do I place this block and couldn't find anything appropriate in server, Will move this to go-kit but again the CIDR ranges need to be configurable because
1. All our integration tests run against a webhook and they might end up failing
2. Any of our load tests we run using an internal webhook would fail

thoughts on this?

Reply:
I agree CIDR needs to be configurable for various reasons. We might want to allowlist  some private/internal IPs or even blocklist a public one in production.